### DISCUSSION TITLE
Copy python executable, not symlink, when creating venv?

### DISCUSSION THREAD

**Discussion by JDLH** [pypa/pipenv#5176] [Q&A]
Is there a way to tell pipenv to copy, rather than make a symlink to, the Python interpreter when creating a virtual environment? 

I have some venvs I set up with pipenv in May 2022, where the `python` executable in the venv's `./bin/` directory is a copy of the original file. Venvs I set up in June and 2022 have a symbolic link from `./bin/python` to the actual file. 

```
% ls -l ~/.local/share/virtualenvs/*/bin/python
lrwxr-xr-x  1 myuser  staff     24  5 Jun 18:30 /Users/myuser/.local/share/virtualenvs/JuneProject-1F3o8dGZ/bin/python -> /opt/local/bin/python3.9
-rwxr-xr-x  1 myuser  staff  34488 10 May 03:55 /Users/myuser/.local/share/virtualenvs/MayProject-FZqwYPDx/bin/python
lrwxr-xr-x  1 jdlh  staff     73 15 Jul 12:26 /Users/myuser/.local/share/virtualenvs/JulyProject-kEV23zV_/bin/python -> /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9
```
My current version of pipenv is 2022.5.2. It is possible that on 10 May 2022 I was using an earlier version of pipenv. 

```
% pipenv --version
pipenv, version 2022.5.2
```

The reason why I want a copy of the interpreter file is that I am using a Python IDE, [PyDev](https://www.pydev.org/manual_101_interpreter.html), which apparently resolves all symbolic links to interpreters into the absolute path to the actual file. And, it lets me use each absolute interpreter path for only one project. Thus when I create multiple venvs which link to the same interpreter executable, it refuses let me configure the subsequent projects to use the venv. I would like it if they changed this behaviour. But until they do, my workaround is to have pipenv copy the Python interpreter executable file, instead of linking to it. 

I am aware that issue #2108 *Virtualenv created includes copy of python exectuable* made essentially the opposite case. I guess there is no one default which will please everyone.

**Comment by matteius** [comment]
@JDLH I believe that the symlinking is done by the virtualenv library when pipenv invokes it to create a non-existent virtualenv.  I am not sure if other options are available when invoking that.

**Reply by ckyoog** [reply]
The option `--copies` of virtualenv is the very option.

**Reply by JDLH** [reply]
Thank you, @ckyoog . So the next questions are, 1) does pipenv have a way to pass the `--copies` option to virtualenv? 2) Does pipenv have a user-accessible option which causes it to pass the `--copies` option to virtualenv, and if so, how does the user access the option?

**Reply by JCapriotti** [reply]
EDIT: I wasn't paying attention and gave an incorrect solution.  The update below is correct.

The only way I was able to do this with pipenv is:

```shell
# Tell pipenv to use a .venv in your project directory
export PIPENV_VENV_IN_PROJECT=1

# Create the virtual environment with the --copies flag
# It appears pipenv will create the .venv directory if it doesn't exist, 
# and will use it as-is if it does exist.
python -m venv --copies .venv

# Continue on with pipenv...
pipenv install --deploy
```
